django/core/cache/backends/base.py

```diff
@@ -160,16 +160,15 @@ def get_or_set(self, key, default, timeout=DEFAULT_TIMEOUT, version=None):
 
         Return the value of the key stored or retrieved.
         """
-        val = self.get(key, version=version)
-        if val is None:
+        val = self.get(key, default, version=version)
```

Comment:
Did you consider using an opaque value here?


```py
UNSET = object()

...

def get_or_set(self, key, default, timeout=DEFAULT_TIMEOUT, version=None):
        val = self.get(key, UNSET, version=version)
        if val is UNSET:
            ...   # handle set case
```

Doing so would mean a `None` isn't treated special.

Reply:
@jdufresne See my [comment](https://github.com/django/django/pull/11812#discussion_r522077756) above and the linked PoC in #13671 that does just what you've suggested.

I'll try to polish it soon. We're also going to deprecate python-memcached as it is unmaintained and doesn't support providing a default.

Reply:
:+1: Nice. Exactly what I had in mind.